gw_spaceheat/actors/home_alone.py

```diff
@@ -194,103 +250,123 @@ async def main(self):
+            # update just_offpeak
+            offpeak: bool = not self.is_onpeak()
+            if offpeak and was_onpeak:
+                just_offpeak = True
+                self.log("Just offpeak")
+            else:
+                just_offpeak = False
+            # and now update was_onpeak for next time
+            was_onpeak = self.is_onpeak()
+
+            # Update top state
+            if self.top_state == HomeAloneTopState.Normal:
+                if self.is_onpeak() and self.is_house_cold():
+                    # implement the change in command tree. Boss: h.n -> h.onpeak-backup
+                    self.set_onpeak_backup_command_tree()
+                    # and let the normal homealone know its dormant until further notice
+                    if self.state != HomeAloneState.Dormant:
+                        self.trigger_normal_event(HomeAloneEvent.GoDormant)
+                    # trigger state change: this will also take the required actions
+                    self.enter_onpeak_backup()
+                    # Report state change to scada
+                    self.top_state_update(cause=TopStateEvent.HouseColdOnpeak)        
+            elif self.top_state == HomeAloneTopState.UsingBackupOnpeak:
+                if just_offpeak:
+                    # trigger state change: this will also take the required actions
+                    self.leave_onpeak_backup()
+                    # Report state change to scada
+                    self.top_state_update()
+                      # implement the change in command tree. Boss: h.onpeak-backup -> h.n
+                    self.set_normal_command_tree()
+                     # and let the normal homealone know its alive 
+                    if self.state == HomeAloneState.Dormant:
+                        self.trigger_normal_event(HomeAloneEvent.WakeUp)
+                                           
+            await self.check_normal_state()
 
-                time_now = datetime.now(self.timezone) 
-                if (((time_now.hour==6 or time_now.hour==16) and time_now.minute>57)
-                    or self.zone_setpoints == {}):
-                    self.get_zone_setpoints()
+            await asyncio.sleep(self.MAIN_LOOP_SLEEP_SECONDS)
 
+    async def check_normal_state(self) -> None:
+        if self.state != HomeAloneState.Dormant:                
+            previous_state = self.state
+
+            if self.is_onpeak():
+                self.storage_declared_ready = False
+                self.full_storage_energy = None
+
+            time_now = datetime.now(self.timezone) 
+            if (((time_now.hour==6 or time_now.hour==16) and time_now.minute>57)
+                or self.zone_setpoints == {}):
+                self.get_zone_setpoints()
+            
+            # Require weather to move out of initializing state
+            # TODO: synth_gen sends message that triggers check_normal_state
+            # when weather arrives
+            if self.weather: 
```

Comment:
This code replaced an async sleep 5. The functional difference is that if the weather isn't around before the first time this core loop runs, it'll wait a minute before making a choice instead of 5 seconds. 

The message-based approach is to get rid of either delay by running check_normal_state on receiving a note from synth_generator (or wherever) that weather now exists.